src/display/dom_utils.js

```diff
@@ -33,6 +33,36 @@ var createValidAbsoluteUrl = sharedUtil.createValidAbsoluteUrl;
 
 var DEFAULT_LINK_REL = 'noopener noreferrer nofollow';
 
+function DOMCanvasFactory() {}
```

Comment:
I'm not sure if using `this` the way that you do here below actually makes sense in the context of a factory!? I'd think that you may need to do something like this instead:
```javascript
function DOMCanvasFactory() {}
DOMCanvasFactory.prototype = {
  create: function DOMCanvasFactory_create(width, height) {
    var canvas = document.createElement('canvas');
    canvas.width = width || 0;
    canvas.height = height || 0;

    return canvas;
  },

  reset: function DOMCanvasFactory_reset(canvas, width, height) {
    if (!canvas) {
      throw new Error('Canvas not found');
    }
    canvas.width = width;
    canvas.height = height;

    return canvas;
  },

  destroy: function DOMCanvasFactory_destroy(canvas) {
    if (!canvas) {
      throw new Error('Canvas not found');
    }
    // Zeroing the width and height causes Firefox to release graphics
    // resources immediately, which can greatly reduce memory consumption.
    canvas.width = 0;
    canvas.height = 0;

    canvas = null;
  }
};
```